---
title: "Microarray technology"
author: "Mikhail Dozmorov"
date: "Fall 2016"
output:
  beamer_presentation:
    colortheme: dolphin
    fig_caption: no
    fig_height: 6
    fig_width: 7
    fonttheme: structurebold
    theme: AnnArbor
  ioslides_presentation:
    css:
    - css/fonts.css
    - css/custom.css
    transition: faster
    widescreen: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

## The Central Dogma of Molecular Biology

DNA is *transcribed* into RNA which is then *translated* into protein
<center>
![](img/Central_Dogma.png)
</center>

## Expression of all genes define phenotypes

- Mendelian genetics explains transmission of genetic information, but sheds no light on how genes create cellular and organismic phenotypes.

- Assays have been developed to more formally study **the association between genes and phenotypes**.

## Rules of base pairing

<center>
![](img/paired_base.png)
</center>

- All genetic code is spelled out with just four chemical letters, or bases: adenine (A), thymine (T), cytosine (C) and guanine (G)
- These pair up, A with T and C with G

## Complementary hybridization

<center>
![](img/paired_base.png)
</center>

- Two single-stranded DNA molecules whose sequences are complementary to each other will exhibit a tendency to bind together to form a single double-stranded DNA molecule. This process is called **hybridization**.

## Complementary hybridization

- Sequence fully complementary to a target will hybridize with much higher efficiency than partially complementary.

<center>
![](img/paired_base2.png)
</center>

- Even when the sequences on the two strands do not match perfectly, as long as there is sufficient **overall** similarity, it is likely that some base pairing will occur.

<center>
![](img/paired_base3.png)
</center>

## Complementary hybridization

- The tendency of DNA strands of complementary sequences to hybridize is exploited in hybridization assays.
- A **probe** consisting of a homogeneous sample of single-stranded DNA molecules, whose sequence is known, is prepared and **labeled with a reporter** fluorescent chemical
- An **immobilized target**, usually a single-stranded DNA molecule, is challenged by the probe.
- As the probes will hybridize preferentially to sequences complementary to the targets, they can be identified by the presence of fluorescence.
- Location of the targets, and the amount of fluorescence, defines which genes, and how much, are expressed.

## What is a Microarray?

- "A DNA microarray is a multiplex technology consisting of thousands of oligonucleotide spots, each containing picomoles of a specific DNA sequence."

- An _oligonucleotide_ (from Greek prefix _oligo_-, "having few, having little") is a short nucleic acid polymer.

## What Are Microarrays Used For? 

- Various molecular assays
<center>
![](img/microarray_used_for.png)
</center>

## What Are Microarrays Used For?

- Biological insights
<center>
![](img/mRNA.png)
</center>

## Microarrays measure expression of all genes

- Traditional molecular biology research followed a "*one gene per experiment*" paradigm
- With the advent of microarrays, research practice has moved from a "one gene at a time" mode to "thousands of genes per experiment" 
- Allows for the study of how genes function _en masse_

## Basic Design of Expression Arrays

- For **each gene** that is a target for the array, we have a **known DNA sequence**
- Microarrays are composed of short DNA sequences complementary to the target genes
- These sequences are attached to a slide at high density 

## Basic Design of Expression Arrays

- mRNA is reverse transcribed to cRNA, and if a complementary sequence is on the on a chip, the cRNA will be more likely to hybridize to it
- The cRNA is labeled with a dye that will fluoresce and generate a signal that is monotonic with the amount of the mRNA sample
- The amount of hybridization can be **quantitatively** measured by the amount of fluorescence

## Microarray types

- Two major types of microarrays
1. **Spotted arrays**, typically _two-channel_
2. **Oligonucleotide arrays**, typically _single-channel_

## Different High-throughput Technology

<center>
![](img/technology.png)
</center>

## Spotted Arrays

- Robotically printed onto a series of glass slides using a robot with needle-heads.
<center>
![](img/spotted_robot.png)
</center>

## Spotted Arrays

- Printing produce a characteristic gridding pattern and almost always use two samples simultaneously (two-color).
<center>
![](img/spotted_array.png)
</center>

## How two-channel arrays work?

- Use two samples, control (reference) and test, e.g., tumor and normal cells.  Take identical amounts of mRNA and convert to cDNA
- Incorporate <span style="font-weight:bold; color:green;">GREEN</span> fluorescent dye into one cDNA (e.g. control)
- Incorporate <span style="font-weight:bold; color:red;">RED</span> fluorescent dye into the other (e.g. test)
- Hybridize mixture of both onto array. <span style="font-weight:bold; color:green;">GREEN</span> spot indicates mRNA expression only in control sample, <span style="font-weight:bold; color:red;">RED</span> - in test sample, <span style="font-weight:bold; color:orange;">ORANGE</span> - in both

## Two-channel (two-color) arrays

<center>
![](img/two_channel_array.png)
</center>

## Combine scans for Red & Green

<center>
![](img/combine_scan.png)
</center>
- False color image is made from digitized fluorescence data, not by superimposing scanned images.

<http://www.bio.davidson.edu/courses/genomics/chip/chip.html>

## Two-channel arrays

- Advantages 
1. Assessment of gene expression in two samples on a single array
2. Two samples have the same background variability on the array
3. Typically, longer molecules are used, so non-specific binding is not much of a problem

## Two-channel arrays 

- Disadvantages
1. More laborious, need to handle two samples
2. Each channel may behave differently
3. Typically, one spot per gene - optical noise is a concern
4. Normalization of microarray data WITHIN and BETWEEN the arrays is still needed

## Single-channel arrays

<center>
![](img/single_chip.png)
</center>

## Single-channel arrays

- mRNA extraction from one sample
- cRNA synthesis and fluorescent dye-labeling
- cRNA hybridization onto array
- Scanning and quantification of fluorescence of each spot

<center>
![](img/single_chip2.png)
</center>

## Oligonucleotide Arrays | Affymetrix arrays

<center>
![](img/Oligonucleotide.png)
</center>

<https://youtu.be/MRmpeBTwwWw>

## RNA Wash

<center>
![](img/RNA_wash.png)
</center>

## RNA Wash

<center>
![](img/RNA_wash2.png)
</center>

## Affymetrix array design

- Rather then an entire gene being placed in a Affymetrix Genechip is an oligonucleotide array consisting of a several **perfect match** (PM) and their corresponding **mismatch** (MM) probes that interrogate for a single gene. 

1. PM probe is the exact complementary sequence of the target genetic sequence, composed of 25 base pairs 

2. MM probe, which has the same sequence with exception that the middle base (13th) position has been reversed 

3. There are roughly 11-20 PM/MM probe pairs that interrogate for each gene, called a probe set 

## Affymetrix array design

<center>
![](img/Affymetrix.png)
</center>

## Affymetrix files

- **DAT** file: Image file, $~10^7$ pixels, ~50 MB.
- **CEL** file: Cell intensity file, probe level PM and MM values.
- **CDF** file: Chip Description File. Describes which probes go in which probe sets and the location of probe-pair sets (genes, gene fragments, ESTs). 

## Alternative splicing

<center>
![](img/alternative_splicing.png)
</center>

## Exon array principles

- Gene-level and exon-level detection of expression.
- Allow detection of alternative splicing mRNA transcripts.

## Exon array design

- PSR - Probe Selection Region
<center>
![](img/exon_array.png)
</center>

## Affymetrix exon arrays

- Affymetrix GeneChip **Exon** 1.0 ST
1. Wide coverage
2. Well annotated genes plus gene prediction sets
3. Over 1.4 million probe sets

## The use of exon array

- Advantages

1. Allow detection of alternative splicing.

2. Cost is about the same as for regular microarrays.

## The use of exon array

- Disadvantages

1. Careful probe design is imperative.

2. Methods for analysis are not well developed.

## Self-Assembling Bead-Arrays

- Made by Illumina
- 3 $\mu$m silicon beads, randomly spread across the surface of the chip
- Each bead coated with ~105 identical 50bp probes
- Each probe has identifying barcode (address) sequences
- ~30 beads per gene

<center>
![](img/bead_array.png)
</center>

## Illumina Bead Arrays

Beads form array on light fibers.

<center>
![](img/beads_from_array.png)
</center>

## Illumina Bead Arrays

Illumination from below excites fluorescence â€“ quantifies probe bound.

<center>
![](img/beads_from_array2.png)
</center>

## Illumina Bead Arrays

- Each chip of the Ref-8 contains 8 arrays with ~ 25,000 targets, plus controls

- Each chip of the WG-6 contains 6 arrays with ~ 50,000 targets, plus controls

- Each chip of the HT-12 chip contains 12 arrays with ~ 50,000 targets and controls

## The use of single-channel arrays

- Advantages

1. Analysis of ONE sample per array

2. Straightforward approach - more fluorescence = more RNA

## The use of single-channel arrays

- Disadvantages

1. Need to use another array(s) for comparative analysis

2. Careful normalization of one microarray data to the other is a must

## InkJet Arrays

<center>
![](img/InkJet.png)
</center>

The new company wanted to determine if **printer** technology could be harnessed to generate microarrays.

***

Inkjet Array Manufacture Involves Sequential Nucleotide Addition

<center>
![](img/InkJet2.png)
</center>

## Copy number alterations (CNA) can lead to disease {.columns-2}

![](img/CNA.png)

- CNAs are a hallmark of tumor genomes

- CNAs can lead to adverse expression changes of affected genes 

- Recurrent CNAs in patients with common phenotype potentially represent molecular markers of disease

- Task: find recurrent CNAs for diagnostics, gene-disease association, disease susceptibility


## Measuring CNAs with array comparative genomic hybridization (aCGH)

- Array hybridization - similar to two-color array studies:
1. Test DNA sample - Unknown DNA copy number
2. Reference DNA sample - normal karyotype DNA copy number
3. Label, mix, hybridize, scan

- Array analysis - resulting data are normalized, log test over reference intensities for genomic targets

## Measuring CNAs with aCGH

- aCGH -  array comparative genomic hybridization
<center>
![](img/CNA2.png)
</center>

## Detecting copy number alterations

<center>
![](img/CNA3.png)
</center>

## Example copy number alterations in cancer

<center>
![](img/CNA4.png)
</center>

## Additional Microarray Platforms

<center>
![](img/additional.png)
</center>

# Applications of microarrays

## All areas of life sciences
- **Cancer research**: Molecular characterization of tumors on a genomic scale; more reliable diagnosis and effective treatment of cancer 

- **Immunology**: Study of host genomic responses to bacterial infections 

- **Model organisms**: Multifactorial experiments monitoring expression response to different treatments and doses, over time or in different cell types 

## Typical comparisons

- Compare mRNA transcript levels 
1. different type of cells, tissues (e.g., liver vs. brain)
2. treatment (Drugs A, B, and C)
3. disease state (tumor vs. normal)
4. different organism (yeast, different strains) different timepoints

## Normal vs. cancerous cells

- All cells in the body are the lineal descendants of a fertilized egg. Almost all of these cells carry genomes that are reasonable accurate copies of the genome that was initially present in the fertilized egg

- However, cells throughout the body are phenotypically distinct (e.g., skin cells versus brain cells) though genetically identical.

- **Differentiation** is the process whereby cells in different parts of the embryo begin to assume distinct phenotypes. 

- The molecular mechanisms of differentiation can be understood by examining the sets of genes that are expressed (transcribed) in some cells but not others. These are tissue-specific genes. 

## Cancer

- Cancer is a disease in which cells escape the restraints on normal cell growth, and become less and less differentiated

- Once a cell has become cancerous, all of its descendant cells are cancerous

- Clonal expansion of cancer cells results in cancer progression

## Hallmarks of cancer

<center>
![](img/cancer.png)
</center>

## Genetic abnormalities in cancer

- Mechanisms whereby mutations and genetic alterations cause cancer 
1. Gain of function (proto-oncogene)
2. Loss of function (tumor suppressor gene)
3. Translocations - creation of chimeric proteins with novel function
4. Aberrant gene expression
5. Epigenetic changes

## Clinical cancer detection

- Pathologist makes an interpretation based upon a compendium of knowledge which may include 
1. Morphological appearance of the tumor
2. Histochemistry
3. Immunophenotyping
4. Cytogenetic analysis 
5. etc. 

<center>
![](img/cancer_detect.png)
</center>

## Microarrays in cancer detection

- Applications of microarrays

1. Characterize molecular variations among tumors by monitoring gene expression 

2. Divide morphologically similar tumors into different groups based on gene expression. 

- Goal: microarrays will lead to more reliable tumor classification and sub-classification (therefore, more appropriate treatments will be administered resulting in improved outcomes) 

